service/src/main/java/com/appirio/service/billingaccount/manager/BillingAccountManager.java

```diff
@@ -624,4 +648,130 @@ private List<ChallengeType> getChallengeTypes() {
         
         return types;
     }
-}
\ No newline at end of file
+
+    /**
+     * Update locked amount for a Challenge of a BillingAccount"
+     *
+     * @return the updated lock Amount
+     */
+    public Float lockAmount(Long billingAccountId, String challengeId, Float requestedLockAmount)  throws SupplyException{
+        //Get the Billing Account
+        List<BillingAccount> originals = getBillingAccount(billingAccountId).getData();
+        if (originals.size() == 0) {
+            throw new SupplyException("Couldn't find billing account with id " + billingAccountId, 404);
+        }
+
+        BillingAccount billingAccount = originals.get(0);
+
+        //Get the Total budget available for the Billing account
+        Float budgetAmount = billingAccount.getBudgetAmount();
+        budgetAmount = budgetAmount == null ? 0 : budgetAmount;
+
+        IdDTO idDto = this.billingAccountDAO.countChallengeIdEntries(billingAccountId, challengeId);
+        Long countChallengeIdEntries = idDto == null ? 0 : idDto.getId();
+
+        FloatDTO floatDto = this.billingAccountDAO.getSumLockedConsumedAmount(billingAccountId, challengeId);
+        Float sumLockedConsumed = floatDto == null ? 0 : floatDto.getFloatValue();
+        sumLockedConsumed = sumLockedConsumed == null ? 0 : sumLockedConsumed;
```

Comment:
Should be `Float sumConsumed = floatDto == null || floatDto.getFloatValue() == null ? 0 : floatDto.getFloatValue();`

Otherwise it could throw NullPointerException: https://stackoverflow.com/questions/3882095/booleans-conditional-operators-and-autoboxing